aries_vcx/Cargo.toml

```diff
@@ -51,7 +51,6 @@ uuid = { version = "0.8", default-features = false, features = ["v4"] }
 strum = "0.16.0"
 strum_macros = "0.16.0"
 derive_builder = "0.10.2"
-tokio = { version = "1.20.4" }
```

Comment:
Why not keep it as a `dev-dependency` though, leaving everything else unchanged?

Reply:
not quite possible yet ... it's a part of bigger problem, and a dilemma @mirgee has run into while doing test refactoring

There's test setup methods in `aries_vcx/src/common/test_utils.rs` such as `create_and_write_test_cred_def` which are used by both:
- tests `#[cfg(test)]` under `aries-vcx/src` - these are often esentially integration tests, but do not yet reside in `aries-vcx/tests`
- integration tests under `aries-vcx/tests`
And for that reason `aries_vcx/src/common/test_utils.rs` cannot be flagged as `#[cfg(test)]` and therefore couldn't access tokio as dev dependency.

So the right solution is to do some further restructuring, move tests around etc. But that would be a different, bigger task than what I am doing here.

Reply:
Okay, I see. 

But in that case I personally disagree with this PR. If everyone else agrees that this is good, feel free to merge it, but I really see this as shoving a problem under the rug. 

I think the proper way that you mentioned is indeed the right solution and should be what gets done.

Reply:
I agree with @bobozaur here, which is the reason why I waited with my review to see what others think about these changes.

Reply:
Fair enough. I'll close and create tracking issue